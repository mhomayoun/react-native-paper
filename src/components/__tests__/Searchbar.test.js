import { render } from '@testing-library/react-native';
import * as React from 'react';
import renderer from 'react-test-renderer';
import Searchbar from '../Searchbar.tsx';

it('renders with placeholder', () => {
  const tree = renderer.create(<Searchbar placeholder="Search" />).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders with text', () => {
  const tree = renderer
    .create(<Searchbar placeholder="Search" value="query" />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('activity indicator snapshot test', () => {
  const tree = renderer.create(<Searchbar loading={true} />).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders with ActivityIndicator', () => {
  const tree = render(<Searchbar loading={true} />);

  expect(tree.getByTestId('activity-indicator')).toBeTruthy();
});

it('renders without ActivityIndicator', () => {
  const tree = render(<Searchbar loading={false} />);

  expect(() => tree.getByTestId('activity-indicator')).toThrow();
});
