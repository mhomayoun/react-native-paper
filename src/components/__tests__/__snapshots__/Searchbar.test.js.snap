// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`activity indicator snapshot test 1`] = `
<View
  collapsable={false}
  style={
    Object {
      "alignSelf": undefined,
      "bottom": undefined,
      "left": undefined,
      "position": undefined,
      "right": undefined,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.15,
      "shadowRadius": 3,
      "top": undefined,
    }
  }
>
  <View
    collapsable={false}
    style={
      Object {
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 1,
      }
    }
  >
    <View
      collapsable={false}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgb(247, 243, 249)",
          "borderRadius": 4,
          "flexDirection": "row",
        }
      }
    >
      <View
        collapsable={false}
        style={
          Object {
            "alignSelf": undefined,
            "bottom": undefined,
            "left": undefined,
            "position": undefined,
            "right": undefined,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
            "top": undefined,
          }
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 0,
                "width": 0,
              },
              "shadowOpacity": 0,
              "shadowRadius": 0,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "backgroundColor": undefined,
                "borderColor": "rgba(121, 116, 126, 1)",
                "borderRadius": 20,
                "borderWidth": 0,
                "elevation": 0,
                "height": 40,
                "margin": 6,
                "overflow": "hidden",
                "width": 40,
              }
            }
          >
            <View
              accessibilityLabel="search"
              accessibilityRole="button"
              accessibilityState={
                Object {
                  "disabled": true,
                }
              }
              accessible={true}
              focusable={false}
              hitSlop={
                Object {
                  "bottom": 6,
                  "left": 6,
                  "right": 6,
                  "top": 6,
                }
              }
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "overflow": "hidden",
                  },
                  Object {
                    "alignItems": "center",
                    "flexGrow": 1,
                    "justifyContent": "center",
                  },
                ]
              }
            >
              <Text
                accessibilityElementsHidden={true}
                importantForAccessibility="no-hide-descendants"
                pointerEvents="none"
                selectable={false}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                    },
                    Object {
                      "color": "rgba(28, 27, 31, 0.54)",
                      "fontSize": 24,
                    },
                  ]
                }
              >
                â–¡
              </Text>
            </View>
          </View>
        </View>
      </View>
      <TextInput
        accessibilityRole="search"
        keyboardAppearance="light"
        placeholder=""
        placeholderTextColor="rgba(28, 27, 31, 1)"
        returnKeyType="search"
        selectionColor="rgba(103, 80, 164, 1)"
        style={
          Array [
            Object {
              "alignSelf": "stretch",
              "flex": 1,
              "fontSize": 18,
              "minWidth": 0,
              "paddingLeft": 8,
              "textAlign": "left",
            },
            Object {
              "color": "rgba(28, 27, 31, 1)",
            },
            undefined,
          ]
        }
        underlineColorAndroid="transparent"
      />
      <View
        accessibilityRole="progressbar"
        accessibilityState={
          Object {
            "busy": true,
          }
        }
        accessible={true}
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
            Object {
              "margin": 10,
            },
          ]
        }
        testID="activity-indicator"
      >
        <View
          collapsable={false}
          style={
            Object {
              "height": 24,
              "opacity": 1,
              "width": 24,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "alignItems": "center",
                "bottom": 0,
                "justifyContent": "center",
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <View
              collapsable={false}
              style={
                Object {
                  "height": 24,
                  "transform": Array [
                    Object {
                      "rotate": "45deg",
                    },
                  ],
                  "width": 24,
                }
              }
            >
              <View
                collapsable={false}
                style={
                  Object {
                    "height": 12,
                    "overflow": "hidden",
                    "width": 24,
                  }
                }
              >
                <View
                  collapsable={false}
                  style={
                    Object {
                      "height": 24,
                      "transform": Array [
                        Object {
                          "translateY": 0,
                        },
                        Object {
                          "rotate": "-165deg",
                        },
                      ],
                      "width": 24,
                    }
                  }
                >
                  <View
                    collapsable={false}
                    style={
                      Object {
                        "height": 12,
                        "overflow": "hidden",
                        "width": 24,
                      }
                    }
                  >
                    <View
                      collapsable={false}
                      style={
                        Object {
                          "borderColor": "rgba(103, 80, 164, 1)",
                          "borderRadius": 12,
                          "borderWidth": 2.4,
                          "height": 24,
                          "width": 24,
                        }
                      }
                    />
                  </View>
                </View>
              </View>
            </View>
          </View>
          <View
            collapsable={false}
            style={
              Object {
                "alignItems": "center",
                "bottom": 0,
                "justifyContent": "center",
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <View
              collapsable={false}
              style={
                Object {
                  "height": 24,
                  "transform": Array [
                    Object {
                      "rotate": "45deg",
                    },
                  ],
                  "width": 24,
                }
              }
            >
              <View
                collapsable={false}
                style={
                  Object {
                    "height": 12,
                    "overflow": "hidden",
                    "top": 12,
                    "width": 24,
                  }
                }
              >
                <View
                  collapsable={false}
                  style={
                    Object {
                      "height": 24,
                      "transform": Array [
                        Object {
                          "translateY": -12,
                        },
                        Object {
                          "rotate": "345deg",
                        },
                      ],
                      "width": 24,
                    }
                  }
                >
                  <View
                    collapsable={false}
                    style={
                      Object {
                        "height": 12,
                        "overflow": "hidden",
                        "width": 24,
                      }
                    }
                  >
                    <View
                      collapsable={false}
                      style={
                        Object {
                          "borderColor": "rgba(103, 80, 164, 1)",
                          "borderRadius": 12,
                          "borderWidth": 2.4,
                          "height": 24,
                          "width": 24,
                        }
                      }
                    />
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`renders with placeholder 1`] = `
<View
  collapsable={false}
  style={
    Object {
      "alignSelf": undefined,
      "bottom": undefined,
      "left": undefined,
      "position": undefined,
      "right": undefined,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.15,
      "shadowRadius": 3,
      "top": undefined,
    }
  }
>
  <View
    collapsable={false}
    style={
      Object {
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 1,
      }
    }
  >
    <View
      collapsable={false}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgb(247, 243, 249)",
          "borderRadius": 4,
          "flexDirection": "row",
        }
      }
    >
      <View
        collapsable={false}
        style={
          Object {
            "alignSelf": undefined,
            "bottom": undefined,
            "left": undefined,
            "position": undefined,
            "right": undefined,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
            "top": undefined,
          }
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 0,
                "width": 0,
              },
              "shadowOpacity": 0,
              "shadowRadius": 0,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "backgroundColor": undefined,
                "borderColor": "rgba(121, 116, 126, 1)",
                "borderRadius": 20,
                "borderWidth": 0,
                "elevation": 0,
                "height": 40,
                "margin": 6,
                "overflow": "hidden",
                "width": 40,
              }
            }
          >
            <View
              accessibilityLabel="search"
              accessibilityRole="button"
              accessibilityState={
                Object {
                  "disabled": true,
                }
              }
              accessible={true}
              focusable={false}
              hitSlop={
                Object {
                  "bottom": 6,
                  "left": 6,
                  "right": 6,
                  "top": 6,
                }
              }
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "overflow": "hidden",
                  },
                  Object {
                    "alignItems": "center",
                    "flexGrow": 1,
                    "justifyContent": "center",
                  },
                ]
              }
            >
              <Text
                accessibilityElementsHidden={true}
                importantForAccessibility="no-hide-descendants"
                pointerEvents="none"
                selectable={false}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                    },
                    Object {
                      "color": "rgba(28, 27, 31, 0.54)",
                      "fontSize": 24,
                    },
                  ]
                }
              >
                â–¡
              </Text>
            </View>
          </View>
        </View>
      </View>
      <TextInput
        accessibilityRole="search"
        keyboardAppearance="light"
        placeholder="Search"
        placeholderTextColor="rgba(28, 27, 31, 1)"
        returnKeyType="search"
        selectionColor="rgba(103, 80, 164, 1)"
        style={
          Array [
            Object {
              "alignSelf": "stretch",
              "flex": 1,
              "fontSize": 18,
              "minWidth": 0,
              "paddingLeft": 8,
              "textAlign": "left",
            },
            Object {
              "color": "rgba(28, 27, 31, 1)",
            },
            undefined,
          ]
        }
        underlineColorAndroid="transparent"
      />
      <View
        collapsable={false}
        style={
          Object {
            "alignSelf": undefined,
            "bottom": undefined,
            "left": undefined,
            "position": undefined,
            "right": undefined,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
            "top": undefined,
          }
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 0,
                "width": 0,
              },
              "shadowOpacity": 0,
              "shadowRadius": 0,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "backgroundColor": undefined,
                "borderColor": "rgba(28, 27, 31, 0.12)",
                "borderRadius": 20,
                "borderWidth": 0,
                "elevation": 0,
                "height": 40,
                "margin": 6,
                "overflow": "hidden",
                "width": 40,
              }
            }
          >
            <View
              accessibilityLabel="clear"
              accessibilityRole="button"
              accessibilityState={
                Object {
                  "disabled": true,
                }
              }
              accessible={true}
              focusable={true}
              hitSlop={
                Object {
                  "bottom": 6,
                  "left": 6,
                  "right": 6,
                  "top": 6,
                }
              }
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "overflow": "hidden",
                  },
                  Object {
                    "alignItems": "center",
                    "flexGrow": 1,
                    "justifyContent": "center",
                  },
                ]
              }
            >
              <Text
                accessibilityElementsHidden={true}
                importantForAccessibility="no-hide-descendants"
                pointerEvents="none"
                selectable={false}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                    },
                    Object {
                      "color": "rgba(28, 27, 31, 0.38)",
                      "fontSize": 24,
                    },
                  ]
                }
              >
                â–¡
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`renders with text 1`] = `
<View
  collapsable={false}
  style={
    Object {
      "alignSelf": undefined,
      "bottom": undefined,
      "left": undefined,
      "position": undefined,
      "right": undefined,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.15,
      "shadowRadius": 3,
      "top": undefined,
    }
  }
>
  <View
    collapsable={false}
    style={
      Object {
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 1,
      }
    }
  >
    <View
      collapsable={false}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgb(247, 243, 249)",
          "borderRadius": 4,
          "flexDirection": "row",
        }
      }
    >
      <View
        collapsable={false}
        style={
          Object {
            "alignSelf": undefined,
            "bottom": undefined,
            "left": undefined,
            "position": undefined,
            "right": undefined,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
            "top": undefined,
          }
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 0,
                "width": 0,
              },
              "shadowOpacity": 0,
              "shadowRadius": 0,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "backgroundColor": undefined,
                "borderColor": "rgba(121, 116, 126, 1)",
                "borderRadius": 20,
                "borderWidth": 0,
                "elevation": 0,
                "height": 40,
                "margin": 6,
                "overflow": "hidden",
                "width": 40,
              }
            }
          >
            <View
              accessibilityLabel="search"
              accessibilityRole="button"
              accessibilityState={
                Object {
                  "disabled": true,
                }
              }
              accessible={true}
              focusable={false}
              hitSlop={
                Object {
                  "bottom": 6,
                  "left": 6,
                  "right": 6,
                  "top": 6,
                }
              }
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "overflow": "hidden",
                  },
                  Object {
                    "alignItems": "center",
                    "flexGrow": 1,
                    "justifyContent": "center",
                  },
                ]
              }
            >
              <Text
                accessibilityElementsHidden={true}
                importantForAccessibility="no-hide-descendants"
                pointerEvents="none"
                selectable={false}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                    },
                    Object {
                      "color": "rgba(28, 27, 31, 0.54)",
                      "fontSize": 24,
                    },
                  ]
                }
              >
                â–¡
              </Text>
            </View>
          </View>
        </View>
      </View>
      <TextInput
        accessibilityRole="search"
        keyboardAppearance="light"
        placeholder="Search"
        placeholderTextColor="rgba(28, 27, 31, 1)"
        returnKeyType="search"
        selectionColor="rgba(103, 80, 164, 1)"
        style={
          Array [
            Object {
              "alignSelf": "stretch",
              "flex": 1,
              "fontSize": 18,
              "minWidth": 0,
              "paddingLeft": 8,
              "textAlign": "left",
            },
            Object {
              "color": "rgba(28, 27, 31, 1)",
            },
            undefined,
          ]
        }
        underlineColorAndroid="transparent"
        value="query"
      />
      <View
        collapsable={false}
        style={
          Object {
            "alignSelf": undefined,
            "bottom": undefined,
            "left": undefined,
            "position": undefined,
            "right": undefined,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
            "top": undefined,
          }
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 0,
                "width": 0,
              },
              "shadowOpacity": 0,
              "shadowRadius": 0,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "backgroundColor": undefined,
                "borderColor": "rgba(121, 116, 126, 1)",
                "borderRadius": 20,
                "borderWidth": 0,
                "elevation": 0,
                "height": 40,
                "margin": 6,
                "overflow": "hidden",
                "width": 40,
              }
            }
          >
            <View
              accessibilityLabel="clear"
              accessibilityRole="button"
              accessibilityState={
                Object {
                  "disabled": false,
                }
              }
              accessible={true}
              focusable={true}
              hitSlop={
                Object {
                  "bottom": 6,
                  "left": 6,
                  "right": 6,
                  "top": 6,
                }
              }
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "overflow": "hidden",
                  },
                  Object {
                    "alignItems": "center",
                    "flexGrow": 1,
                    "justifyContent": "center",
                  },
                ]
              }
            >
              <Text
                accessibilityElementsHidden={true}
                importantForAccessibility="no-hide-descendants"
                pointerEvents="none"
                selectable={false}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                    },
                    Object {
                      "color": "rgba(28, 27, 31, 0.54)",
                      "fontSize": 24,
                    },
                  ]
                }
              >
                â–¡
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
